File: webapp/airTicket/src/app/core/service/user.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {CoreModule} from '../core.module';
import {User} from '../../entity/User';
import {Observable} from 'rxjs';
import {Page} from '../../entity/norm/page';
import {Pageable} from '../../entity/norm/pageable';

@Injectable({
  providedIn: CoreModule
})
export class UserService {

  baseUrl = 'api/user';

  constructor(private httClient: HttpClient) {
  }

  // 登陆
  login(user: User): Observable<User> {
    return this.httClient.post<User>(this.baseUrl + '/login', user);
  }

  // 注销
  logout(): Observable<any> {
    return this.httClient.put(this.baseUrl + '/logout', {});
  }

  // 保存管理员
  saveRoot(user: User): Observable<User> {
    return this.httClient.post<User>(this.baseUrl + '/root', user);
  }

  getAllByUserName(username: string, pageable: Pageable): Observable<Page<User>> {
    const params = {
      username,
      page: pageable.page.toString(),
      size: pageable.size.toString()
    };
    return this.httClient.get<Page<User>>(this.baseUrl + '/getAllByUserName', {params});
  }

  // 删除
  delete(id: number): Observable<any> {
    return this.httClient.delete(this.baseUrl + '/' + id);
  }

  // 重启
  reboot(id: number): Observable<any> {
    return this.httClient.put(this.baseUrl + '/reboot/' + id, {});
  }

  getById(id: number): Observable<User> {
    return this.httClient.get<User>(this.baseUrl + '/' + id);
  }

  resetPassword(id: number, passWord: string): Observable<any> {
    return this.httClient.put(`${this.baseUrl}/resetPassword/${id}`, {passWord});
  }
}

File: api/airTicket/src/main/java/com/xiang/airTicket/service/UserService.java
package com.xiang.airTicket.service;

import com.xiang.airTicket.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import javax.servlet.http.HttpServletResponse;

public interface UserService {

    // 登陆
    User login(String userName, String password);

    // 通过token登陆
    User loginByToken(String userName, String password, HttpServletResponse response);

    // 获取当前登陆用户
    User getCurrentUser();

    // 通过token获取当前登陆用户
    User getCurrentUserByToken(String token);

    // 注册旅客
    User register(User user, HttpServletResponse response);

    // 注销
    void logout();

    // 增加管理员
    User saveRoot(User user);

    // 假删除用户
    void delete(Long id);

    // 修改密码
    void resetPassWord(Long id, String passWord);

    // 获取所有管理员的分页
    Page<User> getAllRoot(Pageable pageable);

    // 通过名字获取所有用户分页
    Page<User> getAllByUserName(String userName, Pageable pageable);

    void reboot(Long id);

    User getById(Long id);

}
